src/Execution/Executor.php

```diff
@@ -104,9 +106,9 @@ public function execute(): ?array
                 ? $this->executeFieldsSerially($objectType, $rootValue, $path, $fields)
                 : $this->executeFields($objectType, $rootValue, $path, $fields);
         } catch (\Throwable $ex) {
-            $this->context->addError(new ExecutionException($ex->getMessage()));
-
-            return [null];
+            $error = new ExecutionException($ex->getMessage(), $fields, null, null, null, null, $ex);
```

Comment:
I just wonder is it worth to create a temp variable here?

Reply:
It's not temporary, it's used twice.

Reply:
@crisu83 In this block, you only add it to handleError?

Reply:
And to add it to the context.

Reply:
Ah true, I moved that into the `handleError` method.

Reply:
Done.

Reply:
@crisu83 Why not just

```
$this->handleError(new ExecutionException(...
```